index.js

```diff
@@ -36,21 +36,22 @@ function serve(root, opts) {
   if (opts.index !== false) opts.index = opts.index || 'index.html';
 
   if (!opts.defer) {
-    return function *serve(next){
-      if (this.method == 'HEAD' || this.method == 'GET') {
-        if (yield send(this, this.path, opts)) return;
+    return function serve(ctx, next){
+      if (ctx.method == 'HEAD' || ctx.method == 'GET') {
+        return send(ctx, ctx.path, opts).then(function(res){
+          return res ? null : next(ctx);
```

Comment:
@coderhaoxin , @tejasmanohar When did @jonathanong change this? As far as I can tell, ctx is a required argument in koa 2.